app/controllers/api/v2/reports_controller.rb

```diff
@@ -62,8 +62,16 @@ def last
 
       private
 
+      def deprecated
+        Foreman::Deprecation.api_deprecation_warning("The resources /reports were moved to /config_reports. Please use the new path instead")
+      end
+
+      def resource_class
+        ConfigReport
+      end
+
       def resource_scope(options = {})
-        super(options).my_reports
+        resource_class.authorized(:view_config_reports).my_reports
```

Comment:
ditto here, can you use super and options to change this, rather than overriding it entirely?